src/approaches/grammar_search_invention_approach.py

```diff
@@ -363,27 +364,27 @@ def _get_successors(s: FrozenSet[Predicate]
 
 
 def _count_positives_for_ops(strips_ops: List[STRIPSOperator],
-                             pruned_atom_data: List[GroundAtomTrajectory]
+                             segments: List[Segment]
                              ) -> Tuple[int, int]:
     """Returns num true positives, num false positives.
     """
     num_true_positives = 0
     num_false_positives = 0
-    for (states, _, atom_sequence) in pruned_atom_data:
-        objects = set(states[0])
+    for segment in segments:
+        objects = set(segment.states[0])
         ground_ops = [o for op in strips_ops
                       for o in utils.all_ground_operators(op, objects)]
-        for i in range(len(atom_sequence)-1):
-            s = atom_sequence[i]
-            ns = atom_sequence[i+1]
-            for ground_op in ground_ops:
-                if not ground_op.preconditions.issubset(s):
-                    continue
-                if ground_op.add_effects == ns - s and \
-                   ground_op.delete_effects == s - ns:
-                    num_true_positives += 1
-                else:
-                    num_false_positives += 1
+        covered_by_some_op = False
+        for ground_op in ground_ops:
+            if not ground_op.preconditions.issubset(segment.init_atoms):
+                continue
+            if ground_op.add_effects == segment.add_effects and \
+               ground_op.delete_effects == segment.delete_effects:
+                covered_by_some_op = True
+            else:
+                num_false_positives += 1
+        if covered_by_some_op:
```

Comment:
Why are we only incrementing once if this segment is covered by any op, rather than incrementing once per op that covers it (as we were doing before)?

Reply:
From a prediction error perspective, I thought this would make more sense. We wouldn't want to give a higher score if we hypothetically had duplicate operators, for example, but the old way would've done that.